driver-core/src/main/com/mongodb/connection/ScramSha1Authenticator.java

```diff
@@ -230,7 +230,13 @@ private String encodeUTF8(final byte[] bytes) throws SaslException {
             }
 
             try {
-                return keyFactory.generateSecret(spec).getEncoded();
+                SecretKey secretKey = keyFactory.generateSecret(spec);
+                // Workaround for https://bugs.openjdk.java.net/browse/JDK-8191177, as suggested in
+                // https://bugs.openjdk.java.net/browse/JDK-8055183
+                //noinspection SynchronizationOnLocalVariableOrMethodParameter
```

Comment:
Does this do anything? Or should they be suppressWarnings?

Reply:
It's what IntelliJ added when I requested it suppress warnings for the statement.  At this language level, annotations are not allowed on synchronized statements